### PR TITLE
Document upgrade guide to 1.0.0

### PR DESCRIPTION
**Pull request by QuentinBisson**
This PR:

* documents the upgrade guide to 1.0.0 https://github.com/giantswarm/roadmap/issues/2879

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [ ] Make sure `values.yaml` and `values.schema.json` are valid. Not needed

### CODE REVIEW

**Review by TheoBrigitte — CHANGES_REQUESTED** [review]

File: docs/upgrade.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+### Upgrade steps
+
+To ensure a smooth upgrade, you can keep the old and new values along side each other before the upgrade to release 20.0.0 and then remove the old values once the upgrade is complete.
+
+Here is the list of changes that you need to apply to your values file:
+
+- Move all user configs from under `apps.<appName>.userConfig` from string to regular helm values to `userConfig.<appName>` if it applies to your use case
+- Rename `userConfig.prometheus-operator-app.configMap.values.prometheus-operator-app` to `userConfig.prometheus-operator-app.configMap.values.kube-prometheus-stack` if it applies to your use case
+- Rename `userConfig.prometheus-operator-app` to `userConfig.kubePrometheusStack` if it applies to your use case
+- Rename `userConfig.promtail-app` to `userConfig.promtail` if it applies to your use case
```
[the remark this answers is not in the document]
Comment (QuentinBisson) [reply docs/upgrade.yaml:26]: Sure thing, I wanted a first eye well first

**Comment by QuentinBisson** [comment]
@TheoBrigitte (as you requested changes) and @marieroque (as you tested the bundle breaking change) could you take a look ? I will not merge until you both approved :)

File: docs/upgrade.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+### Upgrade steps
+
+To ensure a smooth upgrade, you can keep the old and new values along side each other before the upgrade to release 20.0.0 and then remove the old values once the upgrade is complete.
+
+Here is the list of changes that you need to apply to your values file:
+
+- Copy all user configs from `apps.<appName>.userConfig` to regular helm values to `userConfig.<appName>`. Note that this is now moving from a string template to regular values so it is easier to change the values.
+- Rename from `userConfig.prometheus-operator-app.configMap.values.prometheus-operator-app` to `userConfig.kubePrometheusStack.configMap.values.kube-prometheus-stack`
+- Rename `userConfig.promtail-app` to `userConfig.promtail` if it applies to your use case
```
Comment (marieroque) [inline docs/upgrade.md:25]: I'd mention all possible changes:
- prometheus-agent => prometheusAgent
- promtail-app => promtail
- prometheus-operator-app => kubePrometheusStack
- prometheus-operator-crd => prometheusOperatorCrd

File: docs/upgrade.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+The main changes are:
+
+- Moving all user configs from under `apps.<appName>.userConfig` from string to regular helm values to `userConfig.<appName>`
+- Renaming of `prometheus-operator-app` to `kube-prometheus-stack`
+- Renaming of `promtail-app` to `promtail`
+
+### Upgrade steps
+
+To ensure a smooth upgrade, you can keep the old and new values along side each other before the upgrade to release 20.0.0 and then remove the old values once the upgrade is complete.
```
Comment (hervenicol) [inline docs/upgrade.md:19]: This doc talks about upgrading observability bundle to release 1.0.0, and here "upgrade to release 20.0.0" seems out of context.
Don't we have a name for the "Releases"? Like "GiantSwarm Platform  release 20.0.0"?

Comment (QuentinBisson) [reply docs/upgrade.md:19]: Probablye yes

File: docs/upgrade.md (hunk 2 above)
Comment (QuentinBisson) [reply docs/upgrade.md:25]: Yes but also customers are not using it so I did not want to confuse them

Comment (QuentinBisson) [reply docs/upgrade.md:26]: Done

File: docs/upgrade.md (hunk 3 above)
Comment (QuentinBisson) [reply docs/upgrade.md]: DOne thanks

**Review by marieroque — APPROVED** [review]
LGTM

**Comment by QuentinBisson** [comment]
This will merge once @TheoBrigitte approves when he can

**Comment by QuentinBisson** [comment]
Thank you for your reviews :)

File: docs/upgrade.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+```yaml
+apiVersion: v1
+data:
+  values: |
+    apps:
+      prometheus-operator-app:
+        namespace: monitoring
+      kubePrometheusStack:
+        namespace: monitoring
```
Comment (TheoBrigitte) [inline docs/upgrade.md:71]:
```suggestion
      # old values, to remove after upgrade
      prometheus-operator-app:
        namespace: monitoring
      # new values, to add before upgrade
      kubePrometheusStack:
        namespace: monitoring
```

File: docs/upgrade.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+If you are using grafana with persistence enabled, you will need to be careful not to lose your data.
+To that end, you should set the value for both `apps.prometheus-operator-app.userConfig.configMap.values.kube-prometheus-stack.grafana.persistence.existingClaim` and `userConfig.kubePrometheusStack.configMap.values.kube-prometheus-stack.grafana.persistence.existingClaim` to match the name of your existing PVC.
+
+Example:
+
+```yaml
+apiVersion: v1
+data:
+  values: |
+    userConfig:
```
Comment (TheoBrigitte) [inline docs/upgrade.md:89]:
```suggestion
  values: |
    # new values, to add before upgrade
    userConfig:
```

File: docs/upgrade.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+      kubePrometheusStack:
+        configMap:
+          values:
+            kube-prometheus-stack:
+              grafana:
+                persistence:
+                  enabled: true
+                  existingClaim: grafana
+    apps:
```
Comment (TheoBrigitte) [inline docs/upgrade.md:98]:
```suggestion
                  existingClaim: grafana
    # old values, to remove after upgrade
    apps:
```

**Review by TheoBrigitte — APPROVED** [review]

**Pull request merged**